pkg/cache/cache.go

```diff
@@ -65,8 +65,19 @@ type Response struct {
 	// Proxy responds with this version as an acknowledgement.
 	Version string
 
+	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
+	ResourceMarshaled bool
```

Comment:
Who specifies this value? If this is for the go-control-plane library consumer to determine whether to use marshaling, should it be a flag in the Request object / Fetch function instead?

Reply:
For the length check, i think there potentially exist two issues:
- If the resource is unavailable for the specific request, either Resources  or MashaledResources will be empty, should we throw the exception?
- There isn't a good way to add the length check from current design. If adding the length check into the Response, we will need to enforce updating the Resources/MashaledResources via some method, and adding the length check in that method. Another option is to add length check when setting the value to Response, but it happens when implementing the interface https://github.com/envoyproxy/go-control-plane/blob/e78985eb6c01876c02fb837c077253d03b5e6425/pkg/cache/cache.go#L42 and https://github.com/envoyproxy/go-control-plane/blob/e78985eb6c01876c02fb837c077253d03b5e6425/pkg/cache/cache.go#L50, then it is inappropriate since it doesn't enforce the check.

Reply:
Regarding who specifies the value, it happens when go-control-plane library consumers implement the interface https://github.com/envoyproxy/go-control-plane/blob/e78985eb6c01876c02fb837c077253d03b5e6425/pkg/cache/cache.go#L42 and https://github.com/envoyproxy/go-control-plane/blob/e78985eb6c01876c02fb837c077253d03b5e6425/pkg/cache/cache.go#L50.

There shouldn't a flag in the request object, since the actual xDS response sending back to envoy doesn't get changed. 
And the flag will be set when implementing the interface CreateWatch/Fetch.

Reply:
@freedomljc Can you also demonstrate the way it will be set in https://github.com/envoyproxy/go-control-plane/blob/master/pkg/cache/simple.go#L89 so that it will serve as an example?

Reply:
@jyotima , simple.go has implemented the interface Fetch/CreateWatch in: https://github.com/envoyproxy/go-control-plane/blob/2e02e0ea3899d0e6075d62a08fb9647fde055b4d/pkg/cache/simple.go#L168 and https://github.com/envoyproxy/go-control-plane/blob/2e02e0ea3899d0e6075d62a08fb9647fde055b4d/pkg/cache/simple.go#L277.

To support the flag `ResourceMarshaled` in simple.go, we'll need to add the flag in the method Fetch/CreateWatch, and also refactor the logic, then check the flag in https://github.com/envoyproxy/go-control-plane/blob/2e02e0ea3899d0e6075d62a08fb9647fde055b4d/pkg/cache/simple.go#L249 to see whether to marshal the resource before sending back to envoy.

However, just as the filename `simple` indicates, the file is mainly for the entry-level usage of go-control-plane, I'm hesitating to make the change there.